### PR TITLE
[VAULT-2774][VAULT-2775] Add sign and verify to VaultModule 

### PR DESCRIPTION
**Pull request by sharifelgamal**
## Description of changes in release / Impact of release:
(insert text here)

## Documentation
(insert text here)

### Is this a breaking change?
- [ ] Yes
- [ ] No

### If you answered Yes then describe why is it so
(insert text here if applicable)

### Is there a way to disable the change?
- [ ] Use previous release
- [ ] Use a feature flag
- [ ] No

#### Additional details go here
(insert text here if applicable)

### CODE REVIEW

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 1]
```diff
@@ -53,6 +53,12 @@ public class VaultModule implements LarkyVault {
             "VGS_FIXED_LEN_GENERIC"
     );
 
+    private final ImmutableList<String> supposedSigningAlgorithms = ImmutableList.of(
```
Comment (Iapetus999) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:56]: `supposed`?

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 2]
```diff
@@ -188,17 +194,91 @@ public void delete(Object value, Object storage) throws EvalException {
         vault.delete(value, storage);
     }
 
+    @StarlarkMethod(
+        name = "sign",
+        doc = "signs a payload with given keyArn",
+        parameters = {
+            @Param(
+                name = "keyId",
+                doc = "key ARN to sign with",
```
Comment (Iapetus999) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:203]: no key aliases

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 3]
```diff
@@ -188,17 +194,91 @@ public void delete(Object value, Object storage) throws EvalException {
...
+                allowedTypes = {
+                    @ParamType(type = String.class)
+                }
+            ),
+            @Param(
+                name = "algorithm",
+                doc = "the algorithm to use to sign, valid choices are: RSASSA_PSS_SHA_256, RSASSA_PSS_SHA_384, " +
+                    "RSASSA_PSS_SHA_512",
+                allowedTypes = {
+                    @ParamType(type = String.class)
```
Comment (Iapetus999) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:220]: I wonder if there's a way to restrict it

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 4]
```diff
@@ -188,17 +194,91 @@ public void delete(Object value, Object storage) throws EvalException {
...
+    )
+    @Override
+    public Object sign(String keyId, String message, String algorithm) throws EvalException {
+        validateSigningAlgorithm(algorithm);
+        return vault.sign(keyId, message, algorithm);
+    }
+
+    @StarlarkMethod(
+        name = "verify",
+        doc = "verify that a given siganture is valid",
```
Comment (Iapetus999) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:233]: typo

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java (hunk 3 above)
Comment (sharifelgamal) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:220]: we could probably use an enum in a smart way, but i added a validate function so it ends up being the same

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java (hunk 1 above)
Comment (sharifelgamal) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:56]: lmao oops

**Comment by Iapetus999** [comment]
Needs a test 👮

**Comment by sharifelgamal** [comment]
> Needs a test 👮

yep, done

**Review by Iapetus999 — APPROVED** [review]

**Pull request merged**